File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/types/ScenarioDefinition.java
package ai.autonoma.sdk.types;

/**
 * A named scenario (Scenario v2).
 *
 * <p>{@link #up} provisions an isolated environment and returns the data a
 * test needs; the optional {@link #down} tears it back down. Register
 * scenarios on {@link HandlerConfig#getScenarios()}. Implement this interface
 * directly, or build one ergonomically with {@code Scenario.define(...)}.
 */
public interface ScenarioDefinition {

    /** Stable identifier the platform calls up/down by. */
    String name();

    /** Human-readable summary shown in discover. */
    String description();

    /** Runs free-form provisioning code and returns the environment. */
    ScenarioUpResult up(ScenarioUpContext ctx) throws Exception;

    /** Optional teardown. The default is a no-op. */
    default void down(ScenarioDownContext ctx) throws Exception {}
}

File: sdks/php/src/Types/ScenarioDefinition.php
<?php

namespace Autonoma\Sdk\Types;

/**
 * A named scenario. up provisions an isolated environment a test needs; the
 * optional down tears it back down. Register scenarios on
 * HandlerConfig::$scenarios. Build one with Scenario::defineScenario().
 */
final class ScenarioDefinition
{
    public function __construct(
        /** Stable identifier the platform calls up/down by. */
        public readonly string $name,
        /** Human-readable summary shown in discover. */
        public readonly string $description,
        /**
         * @var callable(ScenarioUpContext): (ScenarioUpResult|array<string, mixed>)
         * Free-form provisioning code returning the environment.
         */
        public readonly mixed $up,
        /**
         * @var callable(ScenarioDownContext): void|null
         * Optional teardown; a null down is a no-op.
         */
        public readonly mixed $down = null,
    ) {}
}
